added cast to int so that QSplitter.setSizes works on Python 3.10

NilsNemitz opened the pull request:
While I am not set up to test on 3.10, the error message in #1657 seems quite specific:
`TypeError: index 0 has type 'float' but 'int' is expected`

This adds a cast to int to the parameters in the `QSplitter.setSizes()` call in question.
That might be enough to fix this issue and it doesn't break any other tests.

j9ac9k commented:
Hi @NilsNemitz 

thanks for hoping on this.  doing a squick search in the library, there is potentially a few other places this could happen.

Potentially here: 

https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/canvas/Canvas.py#L116-L118

and potentially here: 

https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/WidgetGroup.py#L22-L24

the second I'm not so sure about.

Would you mind updating the first location as well, and take a glance at the second location and see if you think that's impacted as well?

NilsNemitz commented:
Dear @j9ac9k , I have changed to calculation to integer (and added a cast as a safety net) in the first case.
In the second, the only use of restoreSplitter I can find takes data from splitterState (right above in the code). And it goes through string encoded QByteArray, which should be int enough. 
Although restoreSplitter also accepts a list, I think we don't use that internally. Since this does not seem like a commonly used function, we might be able to leave the responsibility to the user to make sure calls are valid.

Let me know if I missed anything!

j9ac9k commented:
Thanks @NilsNemitz that's the conclusion I came to as well.  This LGTM, merging.
